### ISSUE TITLE
Dynamic joins in aliasing

### ISSUE THREAD

**Issue by PrincessRTFM** [AisteruFire/Velvet#2]
Before I knew that this existed, I was writing my own search tag aliaser with Tampermonkey. The only thing I had that you don't seem to is dynamic joining - my script looked at the last character of any tag it recognized as an alias, and used that to determine whether to join the aliased tags with `&&` or `||`. It didn't replace existing `&&` and `||` in the alias value, it used a magic string (I went with `::`) for that. So, if you defined an alias of `biped` as `human :: anthro` and one of `manes` as `twilight sparkle :: rainbow dash :: fluttershy :: rarity :: pinkie pie :: applejack`, then the tag `biped?` would become `human || anthro` whereas `biped!` became `human && anthro`. But if you had an alias `bipedmanes` that was `(biped?) && (manes?)`, it would become `(human || anthro) && (twilight sparkle || rainbow dash || fluttershy || rarity || pinkie pie || applejack)`.

tl;dr Can you add some kind of magic to allow aliases to expand with either `AND`/`&&` joins or `OR`/`||` joins, based on both an optional indicator in the searched value and a magic string in the alias definition?

**Comment by AisteruFire** [comment]
Can do ! I must admit I'd never have thought of this myself. Just one question : how should it react when `biped` being set to `human :: anthro`, you ask for `biped` without any `?` or `!` at the end ?

**Comment by PrincessRTFM** [comment]
TBH, I'm overhauling my old script for the fun of it right now, and I thought of that same thing. I added a setting to mine for "default magic", so you can specify what to do if there's no indicator. Mine defaults to or-joining, but you can pick whatever default you want. Just let users change it in their settings.

**Comment by AisteruFire** [comment]
At the moment, indicating no `?` or `!` doesn't replace the alias in the query... This would be considered bad, right ?
